packages/docusaurus-preset-classic/src/index.ts

```diff
@@ -74,13 +76,16 @@ export default function preset(
       makePluginConfig('@docusaurus/plugin-google-analytics', googleAnalytics),
     );
   }
-  if (debug || (debug === undefined && !isProd)) {
+  if (isDebugEnabled) {
     plugins.push(require.resolve('@docusaurus/plugin-debug'));
   }
   if (gtag) {
     plugins.push(makePluginConfig('@docusaurus/plugin-google-gtag', gtag));
   }
   if (isProd && sitemap !== false) {
+    if (isDebugEnabled) {
+      sitemap?.ignorePatterns?.push(`/${debugPluginRouteBasePath}/**`);
```

Comment:
If the option doesn't exist, default to `{ ignorePatterns: [] }` so we can still push the debug path?

Reply:
Ah yes, that's right, just fixed it.